### PR TITLE
CVSS 2 editor

### PR DESCRIPTION
**Pull request by domachine**

### CODE REVIEW

File: app/lib/shared/cvss2Tools.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+ */
+export function vectorUpdateFromVectorString(vector) {
+  const { vectorString } = vector
+  if (typeof vectorString !== 'string') return vector
+  const vectorStringMap = vectorString
+    .split('/')
+    .map((k) => {
+      const [key, value] = k.split(':')
+      const mapping = cvss2.mapping.find((m) => m[1] === key)
+      console.log({ mapping })
```
[the remark this answers is not in the document]
Comment (domachine) [reply app/lib/shared/cvss2Tools.js]: Removed :)

File: app/lib/shared/cvss2Tools.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  )
+}
+
+/**
+ * @param {Record<string, string | undefined>} vector
+ * @returns {string}
+ */
+export function vectorGetVectorString(vector) {
+  const vectorMapping = new Map(Object.entries(vector))
+  const f = cvss2.mapping
```
Comment (valentindey) [inline app/lib/shared/cvss2Tools.js:57]: I'd suggest to use a slightly more descriptive name than `f`. Maybe sth. like `vectorFields` would work.

**Comment by valentindey** [comment]
The title and description are missing for all fields in the CVSS2 editor since these are not given in the `cvss-v2.0.json`.

image

In https://github.com/secvisogram/secvisogram/pull/356 I tried to solve that by getting title and description from additional metadata https://github.com/secvisogram/secvisogram/pull/356/commits/9ea2a97e9bd44490774830cb543ffa3aaa89d1c3

**Comment by domachine** [comment]
> The title and description are missing for all fields in the CVSS2 editor since these are not given in the `cvss-v2.0.json`.
> 
> image
> 
> In #356 I tried to solve that by getting title and description from additional metadata [9ea2a97](https://github.com/secvisogram/secvisogram/commit/9ea2a97e9bd44490774830cb543ffa3aaa89d1c3)

I guess this problem will be solved, once we have the translation setup.

File: app/lib/app/SecvisogramPage/View/WizardTab/editors/GenericEditor/CVSS2Editor.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  instancePath,
+  property,
+  parentProperty,
+}) {
+  const { doc, updateDoc, ...outerDocumentEditor } = React.useContext(
+    DocumentEditorContext
+  )
+  const value = instancePath.reduce((value, pathSegment) => {
+    return (value ?? {})[pathSegment]
+  }, /** @type {Record<string, any> | null} */ (doc))
```
[the remark this answers is not in the document]
Comment (domachine) [reply app/lib/app/SecvisogramPage/View/WizardTab/editors/GenericEditor/CVSS2Editor.js:29]: Yes :)

**Review by valentindey — APPROVED** [review]

**Comment by valentindey** [comment]
> > The title and description are missing for all fields in the CVSS2 editor since these are not given in the `cvss-v2.0.json`.
> > image
> > In #356 I tried to solve that by getting title and description from additional metadata [9ea2a97](https://github.com/secvisogram/secvisogram/commit/9ea2a97e9bd44490774830cb543ffa3aaa89d1c3)
> 
> I guess this problem will be solved, once we have the translation setup.

Alright, that's fine for me.

**Pull request merged**